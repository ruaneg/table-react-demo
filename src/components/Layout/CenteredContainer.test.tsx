import { render } from '@testing-library/react';
import 'jest-styled-components';
import { describe, test, expect } from 'vitest';
import { CenteredContainer } from './CenteredContainer';

describe('CenteredContainer', () => {
  test('renders correctly with default styles', () => {
    const { container } = render(<CenteredContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('applies correct styles', () => {
    const { container } = render(<CenteredContainer />);
    const element = container.firstChild;

    expect(element).toHaveStyleRule('display', 'flex');
    expect(element).toHaveStyleRule('justify-content', 'center');
    expect(element).toHaveStyleRule('align-items', 'center');
    expect(element).toHaveStyleRule('height', '100vh');
  });
});
