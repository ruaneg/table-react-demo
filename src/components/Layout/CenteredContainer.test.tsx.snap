// Vitest Snapshot v1

exports[`CenteredContainer > renders correctly with default styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
}

<div
  class="c0"
/>
`;
